cmd/spiffe-aws-assume-role/cli/cli_test.go

```diff
@@ -102,6 +124,58 @@ func TestSetsCustomSessionDuration(t *testing.T) {
 	require.EqualValues(t, 300, *input.DurationSeconds)
 }
 
+func TestSetsCustomStsEndpoint(t *testing.T) {
+	stsEndpoint := uuid.New().String()
+
+	args := []string{
+		"credentials",
+		"--audience=foo",
+		"--role-arn=arn:aws:iam:123456789012:role/foo",
+		fmt.Sprintf("--sts-endpoint=%s", stsEndpoint),
+		"--spiffe-id=spiffe://foo",
+		"--workload-socket=tcp://127.0.0.1:8080",
+	}
```

Comment:
This is our "end-to-end" unit test. It verifies that we can set the STS endpoint and that the value will be plumbed all the way down to the AWS session object which is handed to the STS client.